### PR TITLE
ESB File Structure Documentation

### LINKED ISSUE #23: Write documentation on how all date fields adhere to ISO 8601
## Issue
All date fields in the standard are supposed to adhere to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) (e.g. 2018-01-01), however, that is never spelled out explicitly in the documentation. The only mention of ISO 8601 is in the breakdown of the file naming convention:

> `${ELECTION_DATE}` - The date of the election in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format.

## Solution
Add section of the documentation that talks about the date fields in the preamble to the data field definitions.

### PR DESCRIPTION
**Pull request by JDziurlaj**
resolves #23. This is my first stab at using RST. This document contains a sundry list of best practices and requirements for providing an ESB data format feed. It attempts to be comprehensive, while not overlapping with the existing CSV documentation. I'd open to feedback regarding its structure and content!

### CODE REVIEW

**Comment by jungshadow** [comment]
@JDziurlaj I fixed some rst issues in here and added the new document to the main table of contents. I ran the datapackage parser script and noticed that the `standard.rst.part` file changed. Did you run this when you updated the datapackage? Are you able to [build the docs](https://github.com/csg-org/esb-data-standard#build)? I'm happy to check my version in, but I also want to make sure you're not running into any issues.

**Comment by JDziurlaj** [comment]
I was able to generate the documentation without issue.

File: docs/source/csv/file_structure.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
+ESB Formatting
+==============
+
+.. contents::
+   :local:
+
+ESB File Structure
+------------------
+
+The CSV file must be comma-delimited, UTF-8 (with BOM ommitted) .csv files, named
```
Comment (jungshadow) [inline docs/source/csv/file_structure.rst]:
```suggestion
The CSV file must be comma-delimited, UTF-8 (with BOM omitted) .csv files, named
```

File: docs/source/csv/file_structure.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+according to the `CSV documentation`_.
+
+Header Row
+----------
+
+The first line of the file, or header row, must conform to the following:
+
+- The CSV file must include a header row, containing a comma delimited list of all data fields. NB: All data fields must be included in the header row, even if those fields are never used.
```
Comment (jungshadow) [inline docs/source/csv/file_structure.rst]:
```suggestion
- The CSV file must include a header row containing a comma-delimited list of all data fields. NB: All data fields must be included in the header row, even if those fields are never used.
```

**Pull request merged**